### PR TITLE
[AIR] [Docs] Add "Which preprocessor do I use?" section

### PR DESCRIPTION
**Pull request by bveeramani**
## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [x] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [x] This PR is not tested :(

### CODE REVIEW

File: doc/source/ray-air/preprocessors.rst
Comment (bveeramani) [inline doc/source/ray-air/preprocessors.rst:222]: I feel like this section is a mess. Could use suggestions on how to organize

Comment (dmatrix) [reply doc/source/ray-air/preprocessors.rst:222]: How is this section a mess? Can you elaborate why this subsection is mess?

1. Does it not transition from the previous section?
2. Do you feel we need more text and explanation or scenario to explain why a specific preprocessor should be used?

Comment (richardliaw) [reply doc/source/ray-air/preprocessors.rst:222]: Hmm, it doesn't read grammatically correctly. 

How about -- 
```rst
What preprocessor should I choose?
------------------------------------

The type of preprocessor that you should use depends on the type of data that you have. Below we provide some tips for handling common data formats.

I have Categorical Data Features.
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

If you have categorical data, you may want to convert your data into a vector representation. You can use {A, B, C}  to encode your categorical data into a vector representation. The choice of A, B, or C will depend on the type of data you have.

If your data is a label format, use A to encode your categorical data into a vector representation. Let's take a look at an example:
...
If your data has ordered categories, use B to encode your categorical data into a vector representation. Let's take a look at an example:
...

I have numerical data (floats, ints)
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

.....
```

**Comment by bveeramani** [comment]
Will merge this into #27215 after this has been approved.

**Review by richardliaw — APPROVED** [review]
Looks good! I removed the grouping of the "general preprocessors". It didn't feel like the grouping made that much sense given that this was to be a "problem -> solution" mapping section.

**Comment by bveeramani** [comment]
Closing this PR in favor of #27215

**Comment by dmatrix** [comment]
Sorry @bveeramani I missed reviewing but it seems all sorted.

**Pull request closed without merging**
